Show 'Failed to save map' error message in dialog box

rob-v opened the pull request:
Save map error (e.g. map opened in another program) was ignored so user lost all changes without warning.
Now there is error message and user can fix the issue and save again. Optional changes in ConfirmationDialogs support wider texts (e.g. file paths).

Closes #13036.

Commit:
Fix ignored 'Failed to save map' #13036

obrakmann reviewed (dismissed):
I'm not really happy with the changes to the confirmation dialogs. If you resize the prompt that way, you'd need to align the header, text, and all the buttons as well. Otherwise it just looks unpolished:

![screenshot from 2017-04-20 22-30-05](https://cloud.githubusercontent.com/assets/4331210/25251422/9a4670a4-2619-11e7-94d9-cdf551d47b06.png)

obrakmann commented:
Oh, another note: you don't need to put the issue numbers in the title. Instead, you can put "closes #13036" into the PR description, and it will link the issue to the PR, and also automatically close the issue when the PR is merged. That way you don't need to comment on the issue with the PR's number, either.

rob-v commented:
changes to the confirmation dialogs - ok, I will revert the changes. There shouldn't be such long text, but if they occur, then instead of not aligned labels like in my case, they will overlap (text outside window) so worse, but ok, as you prefer. This was for rare case to improve it.
I won't display the error message, but simple text so it will fit in confirmation window.

ok, thanks for info, I will use Closes..

pchote commented:
The save location is already specified in the main dialog, so IMO it would be more polished to simply show "Access to the file is denied".

rob-v commented:
Updated - I changed it to "Failed to save map '{0}'.".F(map.Title). I saw your comment after commit. Should I change it? The error could be for other reason than access denied.
I didn't explicitly display save location - that path is part of exception message.

pchote commented on OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs:
```diff
@@ -191,19 +191,25 @@ public SaveDirectory(Folder folder, MapClassification classification)
+				}
+				catch (Exception e)
+				{
+					Console.WriteLine("Failed to save map at {0}: {1}", combinedPath, e.Message);
```
I think the best way to do this would be to add `Log.Write("debug", "Failed to save map at {0}: {1}", combinedPath, e.Message);` and then change the dialog text to "See debug.log for details".  This avoids duplication in the UI message and directs map authors to where they can find out what went wrong.

pchote reviewed (changes requested):
The overall approach here is good.  I just have one request to make the error more helpful:

rob-v commented:
Updated

reaperrr commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/51eb5c83d7cf9a1f0aa250d3d3e4a6a2827dbd21...a4fc5ad43c135064d38334dd1d67cefc39a150b3)
